hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/DefaultS3ClientFactory.java

```diff
@@ -175,6 +163,39 @@ public S3AsyncClient createS3AsyncClient(
     return s3AsyncClientBuilder.build();
   }
 
+  private S3AsyncClient createS3CrtAsyncClient(URI uri, S3ClientCreationParameters parameters)
```

Comment:
can we pull all async client creation into its own class, so as to be confident that there's no crt client references elsewhere in the code? I don't what it to become mandatory on the classpath

Reply:
These CRT classes referenced here are all included in the SDK bundle. It's just that if you try to initialise it, and you don't have the CRT dependency on the classpath it'll fail. do you think we still should need to move?